packages/build/src/plugins_core/deploy_config/index.ts

```diff
@@ -0,0 +1,40 @@
+import { promises as fs } from 'fs'
+import { resolve } from 'path'
+
+import merge from 'deepmerge'
+
+import type { NetlifyConfig } from '../../index.js'
+import { CoreStep, CoreStepFunction } from '../types.js'
+
+const coreStep: CoreStepFunction = async function ({ buildDir, netlifyConfig }) {
+  const configPath = resolve(buildDir, '.netlify', 'deploy', 'v1', 'config.json')
+
+  let config: Partial<NetlifyConfig> = {}
+
+  try {
+    const data = await fs.readFile(configPath, 'utf8')
+
+    config = JSON.parse(data) as Partial<NetlifyConfig>
+  } catch {
+    return {}
```

Comment:
Sort of related with the above ☝️ but would be cool to add this error to the span (probably makes sense to create a span for this step in this case though)

Reply:
I've added some logic to ignore the case where we don't find a file and throw an error otherwise.